### PR TITLE
NNS1-3450: Extends csv utility function to support labels

### PR DESCRIPTION
**Pull request by yhabib**
# Motivation

The CSV table header is currently paired with the keys of the object from the array being converted into a CSV. This requires either using "translated" keys or generating a table with very technical names for the header.

# Changes

- Extends `generateCsvFileToSave` and `convertToCsv` to expect a label for the headers so they are decouple from the objects keys.

# Tests

- Added unit test

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary

Prev PR: #5834  | Next PR: #5813

### CODE REVIEW

File: frontend/src/lib/utils/export-to-csv.utils.ts [hunk 1]
```diff
@@ -29,12 +29,12 @@ export const convertToCsv = <T>({
   headers,
 }: {
   data: T[];
-  headers: { id: keyof T }[];
+  headers: { id: keyof T; label?: string }[];
```
Comment (dskloetd) [inline frontend/src/lib/utils/export-to-csv.utils.ts:32]: Let's make it not optional. Not providing a label is just a worse user experience, right?

Comment (yhabib) [reply frontend/src/lib/utils/export-to-csv.utils.ts:32]: I thought we could have some cases where we might not want, eg: we download it for internal use.
But not a strong opinion.

Comment (yhabib) [reply frontend/src/lib/utils/export-to-csv.utils.ts]: I ended up changing it and extracting the types as you recommended in a previous PR

File: frontend/src/tests/lib/utils/export-to-csv.utils.spec.ts [hunk 2]
```diff
@@ -19,90 +23,118 @@ describe("Export to Csv", () => {
...
       ];
       const expected = "formula,value\n'=SUM(A1:A10),100\n\"'+1234567,12\",200";
       expect(convertToCsv({ data, headers })).toBe(expected);
     });
 
     it("should handle values containing newlines by wrapping them in quotes", () => {
-      const data = [
-        { note: "Line 1\nLine 2", id: 1 },
-        { note: "Single Line", id: 2 },
+      const data: TestPersonData[] = [
```
Comment (yhabib) [inline frontend/src/tests/lib/utils/export-to-csv.utils.spec.ts:115]: I changed this one to simplify the number of cases and test types

**Review by dskloetd — APPROVED** [review]

**Pull request merged**